File: Python/1013. Partition Array Into Three Parts With Equal Sum.py
class Solution:
    def canThreePartsEqualSum(self, arr: List[int]) -> bool:
        if sum(arr) % 3 != 0: return False

        avg, temp, counter = sum(arr) // 3, 0, 0

        for i in arr:
            temp += i
            if temp == avg:
                temp = 0
                counter += 1

        return counter >= 3

File: C++/1013. Partition Array Into Three Parts With Equal Sum.cpp
class Solution {
public:
    bool canThreePartsEqualSum(vector<int>& arr) {
        int avg = 0, temp = 0, counter = 0;

        for (int i: arr) avg += i;
        if (avg % 3 != 0) return false;
        avg /= 3;

        for (int i: arr) {
            temp += i;
            if (temp == avg) temp = 0, counter++;
        }

        return counter >= 3;
    }
};
